src/Native/Runtime/StackFrameIterator.cpp

```diff
@@ -569,7 +569,39 @@ void StackFrameIterator::UnwindFuncletInvokeThunk()
 
     bool isFilterInvoke = EQUALS_CODE_ADDRESS(m_ControlPC, RhpCallFilterFunclet2);
 
-#ifdef _TARGET_AMD64_
+#if defined(UNIX_AMD64_ABI)
+    if (isFilterInvoke)
+    {
+        SP = (PTR_UIntNative)(m_RegDisplay.SP);
+        m_RegDisplay.pRbp = SP++;
+    }
+    else
+    {
+        // Save the preserved regs portion of the REGDISPLAY across the unwind through the C# EH dispatch code.
+        m_funcletPtrs.pRbp = m_RegDisplay.pRbp;
+        m_funcletPtrs.pRbx = m_RegDisplay.pRbx;
+        m_funcletPtrs.pR12 = m_RegDisplay.pR12;
+        m_funcletPtrs.pR13 = m_RegDisplay.pR13;
+        m_funcletPtrs.pR14 = m_RegDisplay.pR14;
+        m_funcletPtrs.pR15 = m_RegDisplay.pR15;
+
+        if (EQUALS_CODE_ADDRESS(m_ControlPC, RhpCallCatchFunclet2))
+        {
+            SP = (PTR_UIntNative)(m_RegDisplay.SP + 0x38);
```

Comment:
Would it make sense to declare the local block sizes as constants shared between the asm and C++ code and unify the Unix / Windows code (with two ifdefs around the regdisplay stuff that is platform specific)?
Later we could add target specific methods to the REGDISPLAY to copy the register pointers and to load the registers from the stack - it seems some of the code above could contain less ifdefs if we did that for those too.